packages/api/src/controllers/experiment/attestation.ts

```diff
@@ -0,0 +1,103 @@
+import { Router } from "express";
+import { v4 as uuid } from "uuid";
+import { db } from "../../store";
+import { validatePost } from "../../middleware";
+import { ethers } from "ethers";
+import { makeNextHREF, toStringValues } from "../helpers";
+import _ from "lodash";
+
+const app = Router();
+
+const DOMAIN = {
+  name: "Verifiable Video",
+  version: "1",
+};
+
+app.post("/", validatePost("attestation"), async (req, res) => {
+  const { primaryType, domain, message, signature } = req.body;
+
+  if (primaryType !== "VideoAttestation") {
+    return res.status(400).json({ errors: ["invalid primaryType"] });
+  }
+  if (!_.isEqual(domain, DOMAIN)) {
+    console.log(domain);
+    console.log(DOMAIN);
+    return res.status(400).json({ errors: ["invalid domain"] });
+  }
+  if (!verifySigner(message, signature)) {
+    return res.status(400).json({ errors: ["invalid signature"] });
+  }
+
+  const attestationMetadata = await db.attestation.create({
+    id: uuid(),
```

Comment:
Hmm, what would you think about storing attestations by hash? You can get it from ethers.js [using the TypedDataEncoder interface](https://docs.ethers.org/v5/api/utils/hashing/#TypedDataEncoder); it seems kind of arbitrary to me to have an opaque UUID in here when the process _already_ calculates a unique CID.

Reply:
Yeah, good, idea. We can use the hash of the message. Updated in ce91974a5294a9f5f753a0727a55faa6243276e9. PTAL.